.github/workflows/native-mac.yml

```diff
@@ -0,0 +1,125 @@
+name: Native MacOS
+
+on:
+  push:
+    branches: [ main, ci-sandbox ]
+  pull_request:
+    branches: [ main, ci-sandbox ]
+
+jobs:
+  build-macos:
+    strategy: 
+      fail-fast: false
+      matrix:
+        include:
+          - name: "full-marian"
+            os: macos-10.15
+            cmake: 
+              CMAKE_BUILD_TYPE: "Release"
+              COMPILE_TESTS: "ON"
+              USE_WASM_COMPATIBLE_SOURCE: "OFF"
+              USE_FBGEMM: "OFF"
+              USE_STATIC_LIBS: "OFF"
+              COMPILE_SERVER: "OFF"
+              COMPILE_EXAMPLES: "OFF"
+              TEST_TAGS: ""
+
+          - name: "full-marian-slim"
+            os: macos-10.15
+            cmake: 
+              CMAKE_BUILD_TYPE: "Slim"
+              COMPILE_TESTS: "ON"
+              USE_WASM_COMPATIBLE_SOURCE: "OFF"
+              USE_STATIC_LIBS: "OFF"
+              USE_FBGEMM: "OFF"
+              COMPILE_SERVER: "OFF"
+              COMPILE_EXAMPLES: "OFF"
+              TEST_TAGS: ""
+
+          - name: "minimal-marian"
+            os: macos-10.15
+            cmake: 
+              CMAKE_BUILD_TYPE: "Release"
+              COMPILE_TESTS: "ON"
+              USE_WASM_COMPATIBLE_SOURCE: "OFF"
+              USE_FBGEMM: "OFF"
+              USE_STATIC_LIBS: "OFF"
```

Comment:
`USE_WASM_COMPATIBLE_SOURCE` and `USE_STATIC_LIBS` will be `ON` for custom marian.

Reply:
There's a comment above for mac file complaining about Static libs and it was OFF. My bad regarding the USE_WASM_COMPATIBLE_SOURCE switch. Thanks for pointing out. Will fix that one.

Reply:
```
    # Openblas location is exported explicitly because openblas is keg-only,
    # which means it was not symlinked into /usr/local/.
    # CMake cannot find BLAS on GitHub runners if Marian is being compiled
    # statically, hence USE_STATIC_LIBS=off
```

The above is from source before. I had build failures without this. Although it seems to be working in minimal/custom marian without the need of this I'll try the variations here as well.

Reply:
So, seems this is leading to failures, but not from places expected (blas):
https://github.com/browsermt/bergamot-translator/runs/2333377746
Does this need more investigation?